<a href="https://colab.research.google.com/github/angeliki-tzanou/datasci_5_statistics/blob/main/HHA_507_Assignment_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Assignment #5: Inferential Statistics

##Chi-Square test:

- Loading Packages & dataset:



In [1]:
import pandas as pd

In [5]:
from scipy.stats import chi2_contingency as chi2

In [6]:
df_op = pd.read_csv('opiod.csv')

In [27]:
df_op.head(10)

Year  Patient County Code Patient County Name Rural/Urban       Payer  \
0  2013                  1.0              Albany       Urban    Medicaid   
1  2010                  1.0              Albany       Urban    Medicare   
2  2010                  2.0            Allegany       Rural    Medicaid   
3  2013                  2.0            Allegany       Rural    Medicaid   
4  2010                  2.0            Allegany       Rural       Other   
5  2012                  2.0            Allegany       Rural       Other   
6  2010                  3.0              Broome       Urban  Commercial   
7  2010                  3.0              Broome       Urban    Medicaid   
8  2010                  3.0              Broome       Urban    Medicare   
9  2010                  3.0              Broome       Urban       Other   

   ER Opioid  ER Visits   ER Rate Per 1000   Inpatient ER Opioid  \
0         22       35379              0.622                   15   
1         13       13486              0.964                   13   
2          0        3917              0.000                    2   
3          1        6273              0.159                    1   
4          1         451              2.217                    0   
5          0         560              0.000                    0   
6         24       24971              0.961                   13   
7         14       18615              0.752                   18   
8          6       12025              0.499                    4   
9          0        1103              0.000                    0   

   Inpatient ER Discharges  ...  Ambulatory Surgery Rate Per 1000  \
0                     3408  ...                               0.0   
1                    10785  ...                               0.0   
2                      422  ...                               0.0   
3                      292  ...                               0.0   
4                       59  ...                               0.0   
5                       62  ...                               0.0   
6                     2847  ...                               0.0   
7                     1793  ...                               0.0   
8                     6800  ...                               0.0   
9                      121  ...                               0.0   

   Outpatient Opioid  Outpatient Visits  Outpatient Rate Per 1000  \
0                  0              29976                       0.0   
1                  0                105                       0.0   
2                  0                 26                       0.0   
3                  0               5612                       0.0   
4                  0                 10                       0.0   
5                  0               1010                       0.0   
6                  0                 23                       0.0   
7                  0                 16                       0.0   
8                  0                 20                       0.0   
9                  0                  1                       0.0   

   Other Outpatient Total Opioid   Other Outpatient Total Discharges  \
0                              0                               33921   
1                              0                                9616   
2                              0                                 842   
3                              0                                6388   
4                              0                                 118   
5                              0                                1089   
6                              0                                9576   
7                              0                                1588   
8                              0                                6245   
9                              0                                 162   

   Other Outpatient Total Rate Per 1000  Overall Opioid  Overall Discharges  \
0                        

- Creating the table and the chi-square test:



In [28]:
df_op['Rural/Urban'].value_counts()

Urban    1196
Rural     718
State      36
Name: Rural/Urban, dtype: int64

In [29]:
df_op['Payer'].value_counts()

Medicaid      390
Medicare      390
Other         390
Commercial    390
Unknown       384
All Payers      6
Name: Payer, dtype: int64

In [7]:
contingency_table = pd.crosstab(df_op['Payer'], df_op['Rural/Urban'])

In [8]:
print(contingency_table)

Rural/Urban  Rural  State  Urban
Payer                           
All Payers       0      6      0
Commercial     144      6    240
Medicaid       144      6    240
Medicare       144      6    240
Other          144      6    240
Unknown        142      6    236


- Creating the chi-square test:

In [22]:
from scipy.stats.contingency import chi2_contingency

In [23]:
chi2, p, _, _= chi2_contingency(contingency_table)
print(f"Chi2 value: {chi2}")
print(f"P-value: {p}")

Chi2 value: 319.98611827610137
P-value: 9.183664413295381e-63


- Interpretation of results above:
In this case we can see that the Chi2 value and the P-value are substantially large. The chi2 value being aproximately around 320 signfies a strong correlation between the two variables. Where additionally the high P-value of about 9, represents a strong relationship between the variables of "Payer" & "Rural/Urban" rejecting the possibility of them being independent occurances, thus highlighting the that the type of payer and the area type of residence are stronly related.



## Anova



> Importing packages:



In [37]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

- Ensuring if there needs to be cleaning of the dataset



Checking for missing values:

In [33]:
df_op.columns

Index(['Year', 'Patient County Code', 'Patient County Name', 'Rural/Urban',
       'Payer', 'ER Opioid', 'ER Visits ', 'ER Rate Per 1000 ',
       'Inpatient ER Opioid', 'Inpatient ER Discharges',
       'Inpatient ER Rate Per 1000', 'Inpatient Non ER Opioid',
       'Inpatient Non ER Discharges', 'Inpatient Non ER Rate Per 1000 ',
       'Inpatient Total Opioid', 'Inpatient Total Discharges',
       'Inpatient Total Rate Per 1000 ', 'ER Inpatient Total Opioid',
       'ER Inpatient Total Discharges ', 'ER Inpatient Total Rate Per 1000',
       'Opioid Ambulatory Surgery Visits', 'Ambulatory Surgery Visits',
       'Ambulatory Surgery Rate Per 1000', 'Outpatient Opioid',
       'Outpatient Visits', 'Outpatient Rate Per 1000',
       'Other Outpatient Total Opioid', ' Other Outpatient Total Discharges',
       'Other Outpatient Total Rate Per 1000', 'Overall Opioid',
       'Overall Discharges', 'Overall Rate Per 1000'],
      dtype='object')

In [31]:
missing_data = df_op.isna().sum()
print(missing_data)

Year                                     0
Patient County Code                     36
Patient County Name                      0
Rural/Urban                              0
Payer                                    0
ER Opioid                                0
ER Visits                                0
ER Rate Per 1000                        56
Inpatient ER Opioid                      0
Inpatient ER Discharges                  0
Inpatient ER Rate Per 1000              56
Inpatient Non ER Opioid                  0
Inpatient Non ER Discharges              0
Inpatient Non ER Rate Per 1000          56
Inpatient Total Opioid                   0
Inpatient Total Discharges               0
Inpatient Total Rate Per 1000           56
ER Inpatient Total Opioid                0
ER Inpatient Total Discharges            0
ER Inpatient Total Rate Per 1000        56
Opioid Ambulatory Surgery Visits         0
Ambulatory Surgery Visits                0
Ambulatory Surgery Rate Per 1000         4
Outpatient 

Since the only missing values are in the columns depicting the rate per 1000, it will not affect our calculations thus there is no reason to interfere in those value columns

- Overall cleaning of the formatting of the dataset:

In [34]:
df_op.columns = df_op.columns.str.replace(' ', '_').str.replace('/', '_').str.replace('(', '').str.replace(')', '').str.lower()

<ipython-input-34-3b2cd48398f5>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df_op.columns = df_op.columns.str.replace(' ', '_').str.replace('/', '_').str.replace('(', '').str.replace(')', '').str.lower()
<ipython-input-34-3b2cd48398f5>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df_op.columns = df_op.columns.str.replace(' ', '_').str.replace('/', '_').str.replace('(', '').str.replace(')', '').str.lower()


In [35]:
df_op.columns

Index(['year', 'patient_county_code', 'patient_county_name', 'rural_urban',
       'payer', 'er_opioid', 'er_visits_', 'er_rate_per_1000_',
       'inpatient_er_opioid', 'inpatient_er_discharges',
       'inpatient_er_rate_per_1000', 'inpatient_non_er_opioid',
       'inpatient_non_er_discharges', 'inpatient_non_er_rate_per_1000_',
       'inpatient_total_opioid', 'inpatient_total_discharges',
       'inpatient_total_rate_per_1000_', 'er_inpatient_total_opioid',
       'er_inpatient_total_discharges_', 'er_inpatient_total_rate_per_1000',
       'opioid_ambulatory_surgery_visits', 'ambulatory_surgery_visits',
       'ambulatory_surgery_rate_per_1000', 'outpatient_opioid',
       'outpatient_visits', 'outpatient_rate_per_1000',
       'other_outpatient_total_opioid', '_other_outpatient_total_discharges',
       'other_outpatient_total_rate_per_1000', 'overall_opioid',
       'overall_discharges', 'overall_rate_per_1000'],
      dtype='object')

- Creating the ANOVA table:

Hypothesis: The Payer type, and the area type of residence (Rural or Urban) has a high correlation with ER Opiod visits encountered.

In [38]:
model = ols('er_opioid ~ C(payer) * C(rural_urban)', data=df_op).fit()

In [39]:
anova_table = sm.stats.anova_lm(model, typ=2)
print(anova_table)

                               sum_sq      df             F    PR(>F)
C(payer)                 2.984940e+00     5.0  6.444879e-05  1.000000
C(rural_urban)           6.957661e-04     2.0  3.755627e-08  1.000000
C(payer):C(rural_urban)  2.499140e+03    10.0  2.697986e-02  0.869547
Residual                 1.791460e+07  1934.0           NaN       NaN


/usr/local/lib/python3.10/dist-packages/statsmodels/base/model.py:1888: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 10, but rank is 1
  warnings.warn('covariance of constraints does not have full '


In this case based on the ANOVA table between the er-opiod visits, ruran/urban areas, and payer types do not seem to be highly correlated. All p-values in the above table are rounded to the value of 1 indicating no significant correlation and relationship between the above variables tested.


## T-TEST

In [43]:
from scipy.stats import ttest_ind

In [48]:
from scipy import stats

In [59]:
df_op.columns

Index(['year', 'patient_county_code', 'patient_county_name', 'rural_urban',
       'payer', 'er_opioid', 'er_visits_', 'er_rate_per_1000_',
       'inpatient_er_opioid', 'inpatient_er_discharges',
       'inpatient_er_rate_per_1000', 'inpatient_non_er_opioid',
       'inpatient_non_er_discharges', 'inpatient_non_er_rate_per_1000_',
       'inpatient_total_opioid', 'inpatient_total_discharges',
       'inpatient_total_rate_per_1000_', 'er_inpatient_total_opioid',
       'er_inpatient_total_discharges_', 'er_inpatient_total_rate_per_1000',
       'opioid_ambulatory_surgery_visits', 'ambulatory_surgery_visits',
       'ambulatory_surgery_rate_per_1000', 'outpatient_opioid',
       'outpatient_visits', 'outpatient_rate_per_1000',
       'other_outpatient_total_opioid', '_other_outpatient_total_discharges',
       'other_outpatient_total_rate_per_1000', 'overall_opioid',
       'overall_discharges', 'overall_rate_per_1000'],
      dtype='object')

In [68]:
df_op.head(10)

year  patient_county_code patient_county_name rural_urban       payer  \
0  2013                  1.0              Albany       Urban    Medicaid   
1  2010                  1.0              Albany       Urban    Medicare   
2  2010                  2.0            Allegany       Rural    Medicaid   
3  2013                  2.0            Allegany       Rural    Medicaid   
4  2010                  2.0            Allegany       Rural       Other   
5  2012                  2.0            Allegany       Rural       Other   
6  2010                  3.0              Broome       Urban  Commercial   
7  2010                  3.0              Broome       Urban    Medicaid   
8  2010                  3.0              Broome       Urban    Medicare   
9  2010                  3.0              Broome       Urban       Other   

   er_opioid  er_visits_  er_rate_per_1000_  inpatient_er_opioid  \
0         22       35379              0.622                   15   
1         13       13486              0.964                   13   
2          0        3917              0.000                    2   
3          1        6273              0.159                    1   
4          1         451              2.217                    0   
5          0         560              0.000                    0   
6         24       24971              0.961                   13   
7         14       18615              0.752                   18   
8          6       12025              0.499                    4   
9          0        1103              0.000                    0   

   inpatient_er_discharges  ...  ambulatory_surgery_rate_per_1000  \
0                     3408  ...                               0.0   
1                    10785  ...                               0.0   
2                      422  ...                               0.0   
3                      292  ...                               0.0   
4                       59  ...                               0.0   
5                       62  ...                               0.0   
6                     2847  ...                               0.0   
7                     1793  ...                               0.0   
8                     6800  ...                               0.0   
9                      121  ...                               0.0   

   outpatient_opioid  outpatient_visits  outpatient_rate_per_1000  \
0                  0              29976                       0.0   
1                  0                105                       0.0   
2                  0                 26                       0.0   
3                  0               5612                       0.0   
4                  0                 10                       0.0   
5                  0               1010                       0.0   
6                  0                 23                       0.0   
7                  0                 16                       0.0   
8                  0                 20                       0.0   
9                  0                  1                       0.0   

   other_outpatient_total_opioid  _other_outpatient_total_discharges  \
0                              0                               33921   
1                              0                                9616   
2                              0                                 842   
3                              0                                6388   
4                              0                                 118   
5                              0                                1089   
6                              0                                9576   
7                              0                                1588   
8                              0                                6245   
9                              0                                 162   

   other_outpatient_total_rate_per_1000  overall_opioid  overall_discharges  \
0                        

In [69]:
rural_group = df_op[df_op['rural_urban'] == 'Rural']['er_opioid']
urban_group = df_op[df_op['rural_urban'] == 'Urban']['er_opioid']

In [70]:
t_statistic, p_value = stats.ttest_ind(rural_group, urban_group, equal_var=False)

print(f"T-statistic: {t_statistic}")
print(f"P-value: {p_value}")

T-statistic: -13.07116619617565
P-value: 1.1094279520951892e-36


Based on the values above we can interpret the following. The T-statistic being negative shows how it leans towards the left and further away from the mean, showing that the mean number of er-opioid hopsitilizations for patients in rural areas is lower than the ones for urban areas. The p-value additionally supports that statement due to the fact that its very close to 1 suggesting a signficant difference between the opoiod er occurances from rural and urban areas.

### Regression Analysis:

In [71]:
import matplotlib.pyplot as plt

In [73]:
df_op.head(10)

year  patient_county_code patient_county_name rural_urban       payer  \
0  2013                  1.0              Albany       Urban    Medicaid   
1  2010                  1.0              Albany       Urban    Medicare   
2  2010                  2.0            Allegany       Rural    Medicaid   
3  2013                  2.0            Allegany       Rural    Medicaid   
4  2010                  2.0            Allegany       Rural       Other   
5  2012                  2.0            Allegany       Rural       Other   
6  2010                  3.0              Broome       Urban  Commercial   
7  2010                  3.0              Broome       Urban    Medicaid   
8  2010                  3.0              Broome       Urban    Medicare   
9  2010                  3.0              Broome       Urban       Other   

   er_opioid  er_visits_  er_rate_per_1000_  inpatient_er_opioid  \
0         22       35379              0.622                   15   
1         13       13486              0.964                   13   
2          0        3917              0.000                    2   
3          1        6273              0.159                    1   
4          1         451              2.217                    0   
5          0         560              0.000                    0   
6         24       24971              0.961                   13   
7         14       18615              0.752                   18   
8          6       12025              0.499                    4   
9          0        1103              0.000                    0   

   inpatient_er_discharges  ...  ambulatory_surgery_rate_per_1000  \
0                     3408  ...                               0.0   
1                    10785  ...                               0.0   
2                      422  ...                               0.0   
3                      292  ...                               0.0   
4                       59  ...                               0.0   
5                       62  ...                               0.0   
6                     2847  ...                               0.0   
7                     1793  ...                               0.0   
8                     6800  ...                               0.0   
9                      121  ...                               0.0   

   outpatient_opioid  outpatient_visits  outpatient_rate_per_1000  \
0                  0              29976                       0.0   
1                  0                105                       0.0   
2                  0                 26                       0.0   
3                  0               5612                       0.0   
4                  0                 10                       0.0   
5                  0               1010                       0.0   
6                  0                 23                       0.0   
7                  0                 16                       0.0   
8                  0                 20                       0.0   
9                  0                  1                       0.0   

   other_outpatient_total_opioid  _other_outpatient_total_discharges  \
0                              0                               33921   
1                              0                                9616   
2                              0                                 842   
3                              0                                6388   
4                              0                                 118   
5                              0                                1089   
6                              0                                9576   
7                              0                                1588   
8                              0                                6245   
9                              0                                 162   

   other_outpatient_total_rate_per_1000  overall_opioid  overall_discharges  \
0                        

In [72]:
df_op.columns

Index(['year', 'patient_county_code', 'patient_county_name', 'rural_urban',
       'payer', 'er_opioid', 'er_visits_', 'er_rate_per_1000_',
       'inpatient_er_opioid', 'inpatient_er_discharges',
       'inpatient_er_rate_per_1000', 'inpatient_non_er_opioid',
       'inpatient_non_er_discharges', 'inpatient_non_er_rate_per_1000_',
       'inpatient_total_opioid', 'inpatient_total_discharges',
       'inpatient_total_rate_per_1000_', 'er_inpatient_total_opioid',
       'er_inpatient_total_discharges_', 'er_inpatient_total_rate_per_1000',
       'opioid_ambulatory_surgery_visits', 'ambulatory_surgery_visits',
       'ambulatory_surgery_rate_per_1000', 'outpatient_opioid',
       'outpatient_visits', 'outpatient_rate_per_1000',
       'other_outpatient_total_opioid', '_other_outpatient_total_discharges',
       'other_outpatient_total_rate_per_1000', 'overall_opioid',
       'overall_discharges', 'overall_rate_per_1000'],
      dtype='object')

In [80]:
y = df_op['er_opioid']
x = df_op['er_visits_']

x = sm.add_constant(x)
model = sm.OLS(y, x).fit()
model = sm.OLS(y, x).fit()

print(model.summary())

                            OLS Regression Results                            
Dep. Variable:              er_opioid   R-squared:                       0.896
Model:                            OLS   Adj. R-squared:                  0.895
Method:                 Least Squares   F-statistic:                 1.670e+04
Date:                Sat, 14 Oct 2023   Prob (F-statistic):               0.00
Time:                        01:27:26   Log-Likelihood:                -11553.
No. Observations:                1950   AIC:                         2.311e+04
Df Residuals:                    1948   BIC:                         2.312e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         -0.3802      2.072     -0.183      0.8

In the regression results above we can conclude various obervations. In the R^2 value above shows that there is a stong correlation between those variables. Additionally, the coefficient for the er_visits appears to be slightly positive showing that the number of ER visits is associated with a slight increase in the opioid er occurances. Lastly the p-value is so close to 0 which confirms the statistical signficance existing.